org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintEclipseHeadlessRpcClient.java

```diff
@@ -0,0 +1,185 @@
+      throw new ConfigScopeNotFoundException();
+    }
+    return projectOpt.get();
+  }
+
+  protected Optional<ISonarLintProject> tryResolveProject(String configScopeId) {
+    var projectUri = URI.create(configScopeId);
+    return Stream.of(ResourcesPlugin.getWorkspace().getRoot().findContainersForLocationURI(projectUri))
+      .map(c -> Adapters.adapt(c, ISonarLintProject.class))
+      .filter(Objects::nonNull)
+      .findFirst();
+  }
+
+  protected ConnectionFacade resolveConnection(String connectionId) throws ConnectionNotFoundException {
+    var connectionOpt = SonarLintCorePlugin.getConnectionManager().findById(connectionId);
+    if (connectionOpt.isEmpty()) {
+      SonarLintLogger.get().debug("Unable to resolve connection: " + connectionId);
+      throw new ConnectionNotFoundException();
+    }
+    return connectionOpt.get();
+  }
+
+  @Override
+  public Either<TokenDto, UsernamePasswordDto> getCredentials(String connectionId) throws ConnectionNotFoundException {
+    var connection = resolveConnection(connectionId);
+    return connection.getCredentials();
+  }
+
+  @Override
+  public List<ProxyDto> selectProxies(URI uri) {
+    var proxyService = SonarLintCorePlugin.getInstance().getProxyService();
+    if (proxyService == null) {
+      return List.of();
+    }
+    var proxyDataForHost = proxyService.select(uri);
+    return Arrays.asList(proxyDataForHost).stream()
+      .map(proxyData -> {
+        var proxyType = IProxyData.SOCKS_PROXY_TYPE.equals(proxyData.getType()) ? Proxy.Type.SOCKS : Proxy.Type.HTTP;
+        return new ProxyDto(proxyType, proxyData.getHost(), proxyData.getPort());
+      }).collect(Collectors.toList());
+  }
+
+  @Override
+  public GetProxyPasswordAuthenticationResponse getProxyPasswordAuthentication(String host, int port, String protocol, String prompt, String scheme, URL targetHost) {
+    var proxyService = SonarLintCorePlugin.getInstance().getProxyService();
+    if (proxyService == null) {
+      return new GetProxyPasswordAuthenticationResponse(null, null);
+    }
+    IProxyData[] proxyDataForHost;
+    try {
+      proxyDataForHost = proxyService.select(targetHost.toURI());
+      return Arrays.asList(proxyDataForHost).stream()
+        .findFirst()
+        .map(proxyData -> new GetProxyPasswordAuthenticationResponse(proxyData.getUserId(), proxyData.getPassword()))
+        .orElse(new GetProxyPasswordAuthenticationResponse(null, null));
+    } catch (URISyntaxException e) {
+      SonarLintLogger.get().error("Invalid URL: " + targetHost);
+    }
+    return new GetProxyPasswordAuthenticationResponse(null, null);
+  }
+
+  @Override
+  public boolean checkServerTrusted(List<X509CertificateDto> chain, String authType) {
+    return false;
+  }
+
+  @Override
+  public void didReceiveServerHotspotEvent(DidReceiveServerHotspotEvent params) {
+    // No Hotspots in SLE
+  }
+
+  @Override
+  public void didUpdatePlugins(String connectionId) {
```

Comment:
Isn't it something similar to what we had prior in the `NodeJsManager` implementation?
> SonarLintCorePlugin.getConnectionManager().getConnections().forEach(f -> f.stop());

Reply:
Is it a question for me? :)

Reply:
It was more for myself for later, but maybe you have an answer already 😄

Reply:
Yes, I think stopping all the connected engines is the way to go.